mantis-server/mantis-server-worker/src/main/java/io/mantisrx/server/worker/jobmaster/WorkerMetricHandler.java

```diff
@@ -280,6 +307,28 @@ public void call() {
                                                     gauges.get(ON_NEXT_GAUGE) / 6.0, numWorkers, ""));
                                 }
                             }
+
+                            double sourceJobDrops = 0;
+                            boolean hasSourceJobDropsMetric = false;
+                            Map<String, String> sourceMetricsRecent = sourceJobMetricsRecent.asMap();
+                            for (Map.Entry<String, WorkerMetrics> worker : sourceJobWorkersMap.entrySet()) {
+                                Map<String, GaugeData> metricGroups = metricAggregator.getAggregates(worker.getValue().getGaugesByMetricGrp());
+                                for (Map.Entry<String, GaugeData> group : metricGroups.entrySet()) {
+                                    String metricKey = worker.getKey() + ":" + group.getKey();
+                                    Map<String, Double> gauges = group.getValue().getGauges();
+                                    if (sourceMetricsRecent.containsKey(metricKey) && gauges.containsKey(DROPPED_COUNTER_METRIC_NAME)) {
+                                        sourceJobDrops += gauges.get(DROPPED_COUNTER_METRIC_NAME);
+                                        hasSourceJobDropsMetric = true;
+                                    }
+                                }
+                            }
+                            if (hasSourceJobDropsMetric) {
+                                logger.info("Job stage " + stage + ", recent source job metrics count: " +
+                                        sourceMetricsRecent.size());
+                                jobAutoScaleObserver.onNext(
+                                        new JobAutoScaler.Event(StageScalingPolicy.ScalingReason.SourceJobDrop, stage,
+                                                sourceJobDrops / 6.0 / numWorkers, numWorkers, ""));
```

Comment:
why 6.0?

Reply:
Similar to what's already done on the RPS metric: https://github.com/Netflix/mantis/pull/82/files/e489031772bf021ee83fe0c145e1e202dbc0638a#diff-8068879b4809e7e6458c38f3ab1ce02fb794d943a9b0b4a1953d21d1a07948a4R307

This is to get a per second number and relates to how atlas cumulates metrics for 6 seconds.

Reply:
hmm didn't know that the atlas counters are reset every 6 sec.. should probably add a comment